### PR TITLE
Lab 5 - stringer

### PR DESCRIPTION
**Pull request by loki5265**
Fixes #70

#### Changes proposed in this PR:
* Define IPAddr type
* Implement String() method on IPAddr
* Add test case
* Fix code styles after lint

### CODE REVIEW

File: 05_stringer/lokili/main.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+type IPAddr [4]byte
```
Comment (linwumeng) [inline 05_stringer/lokili/main.go:11]: Prefer to IPv4Addr

File: 05_stringer/lokili/main_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+	"github.com/stretchr/testify/require"
+)
+
+var testSet = []struct {
+	input  IPAddr
+	output string
+}{
+	{IPAddr{192, 168, 31, 1}, "192.168.31.1"},
+	{IPAddr{255, 255, 255, 0}, "255.255.255.0"},
```
Comment (anzboi) [inline 05_stringer/lokili/main_test.go:15]: Would like to see a few more test cases. Try out
`IPAddr{}`
`IPAddr{1, 1}`

File: 05_stringer/lokili/main.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+func (ip IPAddr) String() string {
+	return fmt.Sprintf("%d.%d.%d.%d", ip[0], ip[1], ip[2], ip[3])
```
Comment (anzboi) [inline 05_stringer/lokili/main.go:14]: Perfect.

**Review by anzboi — COMMENTED** [review]
`String` function is good. Would like to see more comprehensive tests.

**Comment by juliaogris** [comment]
Sorry, I will close this PR now in preparation for the next training in a few weeks in Melbourne - feel free to re-open it if you want to continue work on it.

**Pull request closed without merging**